### PR TITLE
Cscap 51 Create Auth, Boot, and Heartbeat responses/requests

### PR DESCRIPTION
**Pull request by cfarhan**
Creates the Authorization, BootNotification, and Heartbeat response/request classes that represent a serialized JSON object.

Creates basic testing for all newly created classes. 

Validation on input values will be done via the MessageValidator, to be done in another ticket. 

Note: Most setters/getters/constructors are handled by Lombok (Annotations above the class declaration).

### CODE REVIEW

File: backend/src/main/java/com/sim_backend/websockets/enums/AuthStatus.java
Comment (pglob) [inline backend/src/main/java/com/sim_backend/websockets/enums/AuthStatus.java:1]: For OCPP field types, use the full name in the filename/classname (i.e. AuthorizationStatus)

File: backend/src/main/java/com/sim_backend/websockets/messages/BootNotificationResponse.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+@OCPPMessageInfo(
+    messageCallID = OCPPMessage.CALL_ID_RESPONSE,
+    messageName = "BootNotificationResponse")
+public class BootNotificationResponse extends OCPPMessage {
+
+  @SerializedName("status")
+  private String status; // Status of the BootNotification (Accepted, Rejected)
+
+  @SerializedName("currentTime")
+  private String currentTime; // Current time in ISO 8601 format
```
Comment (pglob) [inline backend/src/main/java/com/sim_backend/websockets/messages/BootNotificationResponse.java:26]: Can the ZonedDateTime class be used here?

File: backend/src/main/java/com/sim_backend/websockets/enums/BootStatus.java
Comment (pglob) [inline backend/src/main/java/com/sim_backend/websockets/enums/BootStatus.java:1]: Rename to `RegistrationStatus.java`

File: backend/src/main/java/com/sim_backend/websockets/messages/HeartBeatResponse.java [hunk 2]
```diff
@@ -1,11 +1,19 @@
...
+import lombok.AllArgsConstructor;
+import lombok.Getter;
 
-@OCPPMessageInfo(messageName = "HeartBeatResponse", messageCallID = OCPPMessage.CALL_ID_RESPONSE)
+@AllArgsConstructor
+@Getter
+@OCPPMessageInfo(messageCallID = OCPPMessage.CALL_ID_RESPONSE, messageName = "HeartBeatResponse")
 public final class HeartBeatResponse extends OCPPMessage {
 
   /** The HeartBeat's time. */
```
Comment (pglob) [inline backend/src/main/java/com/sim_backend/websockets/messages/HeartBeatResponse.java:19]: Does `currentTime` need something like `@SerializedName("currentTime")`?

**Review by pglob — APPROVED** [review]

File: backend/src/main/java/com/sim_backend/websockets/enums/BootStatus.java
Comment (cfarhan) [reply backend/src/main/java/com/sim_backend/websockets/enums/BootStatus.java:1]: I'd prefer `BootNotificationStatus` for simplicity's sake, will that suffice?

**Pull request merged**